be/src/olap/selection_vector.h

```diff
@@ -0,0 +1,136 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <memory>
+
+#include "util/bitmap.h"
+
+namespace doris {
+
+// Bit-vector representing the selection status of each row in a row block.
+//
+// It is used to scan data with column predicate.
+// the bit in the selection vector is used to indicate whether the given row is live.
+class SelectionVector {
+public:
+    // Construct a new vector. The bits are initially in an indeterminate state.
+    // Call SetAllTrue() if you require all rows to be initially selected.
+    explicit SelectionVector(size_t row_capacity);
+
+    // returen the number of selected rows.
+    size_t count_selected() const;
+
+    // Return true if any rows are selected, or false
+    // This is equivalent to (count_selected() > 0), but faster.
+    bool any_selected() const;
+
+    bool is_row_selected(size_t row) const;
+
+    void set_row_selected(size_t row) {
+        DCHECK_LT(row, _n_rows);
+        BitmapSet(_bitmap.get(), row);
+    }
+
+    void set_all_true() {
+        memset(_bitmap.get(), 0xff, _n_bytes);
```

Comment:
Why not memset _n_bytes + 1, I think it is easier than `pad_extra_bits_wit_zeroes`

Reply:
No, here can not just use memset _n_bytes + 1 to 0xff, the reason is that some calculation is based on byte, and the tail bits should be set to zeroes. 
for example, 

there are 10 rows, so there will be two bytes in the bitmap, and set_all_true will set the prefix 10 bits to 1 and the trailing 6 bits to 0. So the result will be:
11111111 11000000